File: Lecture_04/classwork/passing_arguments/both_ways.f90
! Fortran passes by reference
program both_ways
    implicit none
    real :: a = 0.0
    real :: b = 0.0
    call sub(a, b)
    print "(f3.1, x, f3.1)", a, b
    ! 1.0 0.0
contains
    subroutine sub(x, y)
        ! x is passed by reference,
        ! y is passed by value
        real :: x
        real, value :: y
        x = x + 1.0
        y = y + 1.0
    end subroutine sub
end program both_ways

File: Lecture_04/classwork/passing_arguments/both_ways.c
/* C passes by value */
#include <stdio.h>
void sub(float *x, float y) {
/* for x, its pointer is copied,
   for y, its value is copied */
   *x = *x + 1.0;
    y =  y + 1.0;
}
void main() {
    float a = 0.0;
    float b = 0.0;
    sub(&a, b);
    printf("%.1f %.1f\n", a, b);
    /* 1.0 0.0 */
}
